packages/botonic-react/src/webchat/message-list/index.tsx

```diff
@@ -0,0 +1,106 @@
+import React, { useContext, useEffect, useRef, useState } from 'react'
+
+import { ROLES, WEBCHAT } from '../../constants'
+import { WebchatContext } from '../../contexts'
+import { scrollToBottom } from '../../util'
+import { StyledScrollbar } from '../components/styled-scrollbar'
+import { TypingIndicator } from '../components/typing-indicator'
+import { IntroMessage } from './intro-message'
+import { ScrollButton } from './scroll-button'
+import { ContainerMessage } from './styles'
+import { UnreadMessagesBanner } from './unread-messages-banner'
+
+export const WebchatMessageList = props => {
+  const {
+    webchatState,
+    getThemeProperty,
+    resetUnreadMessages,
+    setLastMessageVisible,
+  } = useContext(WebchatContext)
+
+  const scrollbarOptions = {
+    ...{ enable: true, autoHide: true },
+    ...getThemeProperty(WEBCHAT.CUSTOM_PROPERTIES.scrollbar),
+  }
+
+  const [firstUnreadMessageId, setFirstUnreadMessageId] = useState()
+  useEffect(() => {
+    const firstUnreadMessage = webchatState.messagesComponents.find(
+      message => message.props.isUnread
+    )
+    setFirstUnreadMessageId(firstUnreadMessage?.props?.id)
+  }, [webchatState.messagesComponents])
+
+  const lastMessageRef = useRef(null)
+  useEffect(() => {
+    if (webchatState.messagesComponents.length > 0 && lastMessageRef.current) {
+      const observer = new IntersectionObserver(entries => {
+        entries.forEach(entry => {
+          setLastMessageVisible(entry.isIntersecting)
+        })
+      })
+      observer.observe(lastMessageRef.current)
+    }
+  }, [webchatState.messagesComponents])
+
+  // TODO
+  useEffect(() => {
+    if (webchatState.isLastMessageVisible && webchatState.typing) {
+      scrollToBottom({ host: props.host })
+    }
+    if (webchatState.isLastMessageVisible) {
+      scrollToBottom({ host: props.host })
+    }
+  }, [webchatState.typing, webchatState.isLastMessageVisible])
+
+  const handleScrollToBottom = () => {
+    resetUnreadMessages()
+    scrollToBottom({ host: props.host })
+  }
+
+  const showUnreadMessagesBanner = (messageComponentId: string) =>
+    firstUnreadMessageId &&
+    messageComponentId === firstUnreadMessageId &&
+    webchatState.unreadMessages > 0
```

Comment:
Shouldn't this function also check if the enable prop inside notifications prop is not false? I think it should be checked here and not inside the component... Otherwise this method can return true but you are not showing thebanner because enable prop is set to false...

Reply:
I don't see it the same way. The MessagesList component I think should not have the logic of whether the UnreadMessageBanner component is activated or if it is a custom (since you can activate it both ways), this is part of the logic of the UnreadMessageBanner.